README.md

```diff
@@ -165,78 +184,566 @@ and then destroy them, all from the command line:
 puppet resource ec2_securitygroup test-group ensure=absent region=sa-east-1
 ```
 
+### Managing AWS Infrastructure
+
+You can use the aws module to audit AWS resources, launch autoscaling groups in VPC, perform unit testing, and more. The [examples](examples/) directory contains a variety of usage examples that should give you an idea of what's possible:
+
+* [Puppet Enterprise](puppet-enterprise/): Start up a small Puppet Enterprise cluster using the AWS module.
+* [Managing DNS](manage-dns/): Manage DNS records in Amazon Route53 using Puppet.
+* [Data Driven Manifests](data-driven-manifests/): Automatically generate resources based on a data structure.
+* [Hiera Example](hiera-example/): Store common information like region or AMI id in Hiera.
+* [Infrastructure as YAML](yaml-infrastructure-definition/): Describe an entire infrastructure stack in YAML, and use `create_resources` and Hiera to build your infrastructure.
+* [Auditing Resources](audit-security-groups/): Audit AWS resource changes and work alongside other tools.
+* [Unit Testing](unit-testing): Test your AWS code with Puppet testing tools like rspec-puppet.
+* [Virtual Private Cloud](vpc-example): Use the Puppet DSL to manage a AWS VPC environment.
+* [Using IAM permissions](iam-profile): Control the API permissions required by the module with an IAM profile.
+* [Elastic IP Addresses](elastic-ip-addresses/): Attach existing elastic IP addresses to instances managed by Puppet
+  
 ##Reference
 
-The following shows each of the new types along with a full list of
-their parameters.
-
-### ec2_instance
-
-```puppet
-ec2_instance { 'name-of-instance':
-  ensure            => present,
-  region            => 'us-east-1',
-  availability_zone => 'us-east-1a',
-  image_id          => 'ami-123456',
-  instance_type     => 't1.micro',
-  monitoring        => true,
-  key_name          => 'name-of-existing-key',
-  security_groups   => ['name-of-security-group'],
-  user_data         => template('module/file-path.sh.erb'),
-  tags              => {
-    tag_name => 'value',
-  },
-}
-```
+### Types
```

Comment:
This section is so long it feels like at best it needs an index. I wonder if this would be better as separate files in a directory, with links here in the README.